homeassistant/components/lametric/__init__.py

```diff
@@ -1,52 +1,82 @@
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.typing import ConfigType
 
-from .const import DOMAIN, LOGGER
+from .const import DOMAIN
 
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_CLIENT_ID): cv.string,
-                vol.Required(CONF_CLIENT_SECRET): cv.string,
-            }
-        )
-    },
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_CLIENT_ID): cv.string,
+                    vol.Required(CONF_CLIENT_SECRET): cv.string,
+                }
+            )
+        },
+    ),
     extra=vol.ALLOW_EXTRA,
 )
 
 
 def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the LaMetricManager."""
-    LOGGER.debug("Setting up LaMetric platform")
-    conf = config[DOMAIN]
-    hlmn = HassLaMetricManager(
-        client_id=conf[CONF_CLIENT_ID], client_secret=conf[CONF_CLIENT_SECRET]
-    )
-    if not (devices := hlmn.manager.get_devices()):
-        LOGGER.error("No LaMetric devices found")
-        return False
-
-    hass.data[DOMAIN] = hlmn
-    for dev in devices:
-        LOGGER.debug("Discovered LaMetric device: %s", dev)
+    """Set up the LaMetric integration."""
+    if DOMAIN in config:
+        async_create_issue(
+            hass,
+            DOMAIN,
+            "manual_migration",
+            breaks_in_ha_version="2022.9.0",
+            is_fixable=False,
+            severity=IssueSeverity.ERROR,
+            translation_key="manual_migration",
+        )
 
     return True
 
 
-class HassLaMetricManager:
-    """A class that encapsulated requests to the LaMetric manager."""
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up LaMetric from a config entry."""
+    lametric = LaMetricDevice(
+        host=entry.data[CONF_HOST],
+        api_key=entry.data[CONF_API_KEY],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        device = await lametric.device()
+    except LaMetricConnectionError as ex:
+        raise ConfigEntryNotReady("Cannot connect to LaMetric device") from ex
 
-    def __init__(self, client_id: str, client_secret: str) -> None:
-        """Initialize HassLaMetricManager and connect to LaMetric."""
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = lametric
 
-        LOGGER.debug("Connecting to LaMetric")
-        self.manager = LaMetricManager(client_id, client_secret)
-        self._client_id = client_id
-        self._client_secret = client_secret
+    # Set up notify platform, no entry support for notify component yet,
+    # have to use discovery to load platform.
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            Platform.NOTIFY,
+            DOMAIN,
+            {CONF_NAME: device.name, "entry_id": entry.entry_id},
+            {},
```

Comment:
We need to pass the original config from `async_setup` here as last parameter to `discovery.async_load_platform`.

Store the config in hass.data. See eg the webostv integration.

Reply:
Which original config? There is none?

Reply:
Not for this integration, but `discovery.async_load_platform` expects the whole original config of Home Assistant.